lnrpc/signrpc/signer.proto

```diff
@@ -265,3 +301,125 @@ message SharedKeyResponse {
     // The shared public key, hashed with sha256.
     bytes shared_key = 1;
 }
+
+message MuSig2SessionRequest {
+    /*
+    The key locator that identifies which key to use for signing.
+    */
+    KeyLocator key_loc = 1;
+
+    /*
+    A list of all public keys (serialized in 33-byte compressed format)
+    participating in the signing session. The list will always be sorted
+    lexicographically internally. This must include the local key which is
+    described by the above key_loc.
+    */
+    repeated bytes all_signer_pubkeys = 2;
+
+    /*
+    An optional list of all public nonces of other signing participants that
+    might already be known.
+    */
+    repeated bytes other_signer_public_nonces = 3;
+}
+
+message MuSig2SessionResponse {
+    /*
+    The unique ID that represents this signing session. A session can be used
+    for producing a signature a single time. If the signing fails for any
+    reason, a new session with the same participants needs to be created.
+    */
+    bytes session_id = 1;
+
+    /*
+    The combined public key.
+    */
+    bytes combined_key = 2;
+
+    /*
+    The two public nonces the local signer uses, combined into a single value
+    of 66 bytes. Can be split into the two 33-byte points to get the individual
+    nonces.
+    */
+    bytes local_public_nonces = 3;
```

Comment:
FWIW this is only actually needed for incremental verification, but maybe it's also useful on the caller side of things?

Reply:
We need to communicate the nonce to the other party, so we need to know it at some point, don't we? Or how would the nonce exchange work?